src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +831,149 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
             "details": errors_by_operation,
             "message": message
         }
+
+    def __set_error_detail(self, error_code, message):
+        formatted_message = self.__ensure_error_message_restriction_compliance(message)
+        return {
+            "code": str(error_code),
+            "message": str(formatted_message)
+        }
     # endregion
 
+    # region - Patch Truncation
+    def get_num_assessment_patches_removed(self):
+        return len(self.__assessment_patches_removed)
+
+    def get_num_installation_patches_removed(self):
+        return len(self.__installation_patches_removed)
+
+    def log_truncated_patches(self):
+        """ log details of all the removed patches from status """
+        self.composite_logger.log_debug("Count of patches removed from: [Assessment={0}] [Installation={1}]".format(self.get_num_assessment_patches_removed(), self.get_num_installation_patches_removed()))
+
+    def __get_status_payload_with_truncated_patches(self, status_file_payload_json_dumps):
+        """ Get truncated status file payload when status file byte size is more than 126kb """
+        status_file_size_in_bytes = self.__calc_status_size_on_disk(status_file_payload_json_dumps)  # calc complete_status_file_payload_json byte size on disk
+
+        if status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:  # perform truncation complete_status_file byte size > 126kb
+            truncated_status_file = self.__create_truncated_status_file(status_file_size_in_bytes, status_file_payload_json_dumps)
+            status_file_payload_json_dumps = json.dumps(truncated_status_file)
+        return status_file_payload_json_dumps
+
+    def __create_truncated_status_file(self, status_file_size_in_bytes, complete_status_file_payload_json):
+        """ Truncate substatus message patches when complete status file size is greater than 126kb """
+        self.composite_logger.log_debug("Begin patches truncation: [StatusFileSizeInBytes={0}] [InternalFileSizeLimitInBytes={1}]".format(
```

Comment:
Use log_verbose in all high-volume code paths.

Reply:
done